File: backend/src/main/java/com/app/mevocab/internal/user/dto/response/LoginResponse.java
package com.app.mevocab.internal.user.dto.response;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class LoginResponse {
    String userID;
    String name;
    String token;
}

File: frontend/lib/data/models/dto/Response/LoginResponse.dart
class LoginResponse {
  final String userID;
  final String name;
  final String token;

  LoginResponse({
    required this.userID,
    required this.name,
    required this.token,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      userID: json['userID'] ?? '',
      name: json['name'] ?? '',
      token: json['token'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userID': userID,
      'name': name,
      'token': token,
    };
  }
}
